core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/TupleExprBuilder.java

```diff
@@ -1855,7 +1855,7 @@ public MathExpr visit(ASTMath node, Object data) throws VisitorException {
 	public Object visit(ASTFunctionCall node, Object data) throws VisitorException {
 		ValueConstant uriNode = (ValueConstant) node.jjtGetChild(0).jjtAccept(this, null);
 		IRI functionURI = (IRI) uriNode.getValue();
-		if (CustomAggregateFunctionRegistry.getInstance().has(functionURI.stringValue())) {
+		if (CustomAggregateFunctionRegistry.getInstance().has(functionURI.stringValue()) || node.isDistinct()) {
```

Comment:
I was testing if there was a way to not have to rely on the CustomAggregateFunctionRegistry during parsing. I ran into some issues in some of the other code. Maybe it's cleanest to just remove my change.